sci-misc/kaldi/kaldi-0_p20160628.ebuild

```diff
@@ -0,0 +1,133 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit toolchain-funcs cuda flag-o-matic
+
+COMMIT_ID="16e69f1aacce8ad0665d2b6666c053b1421a9e91"
+DESCRIPTION="A toolkit for speech recognition"
+HOMEPAGE="http://kaldi-asr.org/"
+SRC_URI="https://github.com/kaldi-asr/kaldi/archive/${COMMIT_ID}.tar.gz -> ${P}.tar.gz"
+
+S=${WORKDIR}/${PN}-${COMMIT_ID}/src
+
+LICENSE="Apache-2.0"
+SLOT="0"
+IUSE="doc test cuda speex"
+KEYWORDS="~amd64"
+
+RDEPEND="
+	speex? ( media-libs/speex )
+	virtual/cblas
+	virtual/lapack
+	virtual/lapacke
+	<sci-misc/openfst-1.5.0
+	cuda? ( dev-util/nvidia-cuda-toolkit )"
+DEPEND="${RDEPEND}
+	doc? ( app-doc/doxygen )"
+
+# We need write acccess /dev/nvidiactl, /dev/nvidia0 and /dev/nvidia-uvm and the portage
+# user is (usually) not in the video group
+RESTRICT="test? ( cuda? ( userpriv ) )"
+
+pkg_pretend() {
+	local cblas_provider=$(eselect cblas show)
+
+	if [[ ! ${cblas_provider} =~ (atlas|mkl|openblas) ]]; then
+		die "Build with '${cblas_provider}' CBLAS is not supported"
+	fi
+}
+
+src_prepare() {
+	eapply \
+		"${FILESDIR}"/Makefile.patch \
+		"${FILESDIR}"/default_rules.mk.a7d9824.patch
+	eapply_user
+}
+
+src_configure() {
+	append-cxxflags \
+		-DHAVE_EXECINFO_H=1 \
+		-DHAVE_CXXABI_H \
+		-DHAVE_OPENFST_GE_10400 \
+		-std=c++11 \
+		-Wall \
+		-I.. \
+		-pthread \
+		-Wno-sign-compare \
+		-Wno-unused-local-typedefs \
+		-Winit-self \
+		-rdynamic \
+		-fPIC \
+		$($(tc-getPKG_CONFIG) --cflags cblas) \
+		$($(tc-getPKG_CONFIG) --cflags lapack)
+
+	append-libs \
+		-lfst \
+		-lm \
+		-lpthread \
+		-ldl \
+		$($(tc-getPKG_CONFIG) --libs cblas) \
+		$($(tc-getPKG_CONFIG) --libs lapack)
+
+	local cblas_provider=$(eselect cblas show)
+
+	if [[ ${cblas_provider} =~ atlas ]]; then
+		append-cxxflags -DHAVE_ATLAS
+	elif [[ ${cblas_provider} =~ mkl ]]; then
+		append-cxxflags -DHAVE_MKL
+	elif [[ ${cblas_provider} =~ openblas ]]; then
+		append-cxxflags -DHAVE_OPENBLAS $($(tc-getPKG_CONFIG) --cflags lapacke)
+	fi
+
+	use test || append-cxxflags -DNDEBUG
+
+	if use speex; then
+		append-cxxflags -DHAVE_SPEEX
+		append-libs -lspeex
+	fi
+
+	cat <<-EOF > kaldi.mk
+		KALDI_FLAVOR := dynamic
+		KALDILIBDIR := "${S}"/lib
+		CC = $(tc-getCXX)
+		RANLIB = $(tc-getRANLIB)
+		LDLIBS = ${LIBS}
+		DOUBLE_PRECISION = 0
```

Comment:
`DOUBLE_PRECISION = 0` is restored because it is required by CUDA specific Makefile.